core/src/test/java/org/infinispan/stream/DistributedStreamIteratorWithStoreAsBinaryTest.java

```diff
@@ -41,6 +41,7 @@ protected void createCacheManagers() throws Throwable {
       builderUsed.clustering().hash().numOwners(1);
       builderUsed.memory().storageType(StorageType.BINARY);
       createClusteredCaches(3, builderUsed);
+      cacheManagers.forEach(cm -> cm.getClassWhiteList().addRegexps("java.util.Collections\\$SingletonMap"));
```

Comment:
We should add `SingletonMap` to the `CollectionExternalizer`

Reply:
That's won't apply to this use-case, because the filter provided by the user (test) relies on Serializable and so both the filter and the map are handled by the user marshaller.

Reply:
So I am confused, doesn't the user have some way of invoking our internal Marshaller classes? Or do they have to support Serialize every single class they may use?

Reply:
In general they shouldn't be using our internal classes. But it's possible todo so if they provide  externalizers for their classes.